importer/tag/src/test/java/org/itsallcode/openfasttrace/importer/tag/TestTagImporter.java

```diff
@@ -20,353 +20,129 @@
-            final SpecificationItemId coveredId, final List<String> neededArtifactTypes)
-    {
-        final SpecificationItemId generatedId = SpecificationItemId.createId(artifactType,
-                coveredId.getName(), 0);
-        final Builder itemBuilder = SpecificationItem.builder() //
-                .id(generatedId) //
-                .addCoveredId(coveredId) //
-                .location(FILENAME, lineNumber);
-        neededArtifactTypes.forEach(itemBuilder::addNeedsArtifactType);
-        return itemBuilder.build();
-    }
-
-    private static SpecificationItem item(final String artifactType, final int lineNumber,
-            final int counter, final SpecificationItemId coveredId, final List<String> neededArtifactTypes)
-    {
-        final SpecificationItemId generatedId = SpecificationItemId.createId(artifactType,
-                generateName(coveredId, lineNumber, counter), 0);
-        final Builder itemBuilder = SpecificationItem.builder() //
-                .id(generatedId) //
-                .addCoveredId(coveredId) //
-                .location(FILENAME, lineNumber);
-        neededArtifactTypes.forEach(itemBuilder::addNeedsArtifactType);
-        return itemBuilder.build();
-    }
-
-    private static String generateName(final SpecificationItemId coveredId, final int lineNumber,
-            final int counter)
-    {
-        final String uniqueName = FILENAME + lineNumber + counter + coveredId.toString();
-        final CRC32 checksum = new CRC32();
-        checksum.update(uniqueName.getBytes(StandardCharsets.UTF_8));
-        return coveredId.getName() + "-" + checksum.getValue();
-    }
-
-    private void assertItems(final String content, final SpecificationItem... expectedItems)
-    {
-        final List<SpecificationItem> actual = runImporter(content);
-        assertThat(actual, hasSize(expectedItems.length));
-        if (expectedItems.length > 0)
+    static Stream<Arguments> tagImporterTests()
+    {
+        return Stream.of(
+                noItemDetected(""),
+                noItemDetected("non empty string"),
+                noItemDetected("non empty string\nmultiline"),
+                noItemDetected("impl->dsn~missing-brackets~1"),
+                noItemDetected("[missing arrow]"),
+                noItemDetected("[impl->invalid covered tag]"),
+                noItemDetected("[impl->dsn~missing-revision]"),
+                noItemDetected("[impl->dsn~invalid-revision~invalid]"),
+                noItemDetected("[impl->\ndsn~newline-after-arrow~1]"),
+                noItemDetected("[impl->dsn~name1~1>>]"),
+                noItemDetected("[impl->dsn~name1~1>>tag with space]"),
+
+                parsedItem("[impl->dsn~name1~1" + "]", item("impl~name1-912633853~0", "dsn~name1~1")),
+                parsedItem("[ impl -> dsn~name1~1 " + "]", item("impl~name1-912633853~0", "dsn~name1~1")),
```

Comment:
I suggest you switch the item parts up now and then. Otherwise the item recognition could just be a constant. :-)